src/util/stream-to-promise.js

```diff
@@ -0,0 +1,9 @@
+/* @flow */
+
+export async function streamToPromise(
```

Comment:
Oh wow, good catch. I would suggest filing a bug upstream when you get a chance.

For `web-ext` purposes, we need the error handling and test coverage I mentioned here. However, if that feels like a big undertaking maybe you can just add `skip()` to the relating tests and file a `web-ext` issue to add our own `streamToPromise`.